### ISSUE TITLE
undefined method stack trace

### ISSUE THREAD

**Issue by tomkadwill** [lotus/view#61]
The following view class has a method `a_method`, which is valid. Inside there is a method `this_method_raises_an_error` which raises an error.

```
module Web::Views::Home
  require 'lotus/helpers'

  class Index
    include Web::View

    def a_method
      this_method_raises_an_error
    end
  end
end
```
When calling `a_method` from the template, The resulting stack trace is not very specific and doesn't point to the problem function `this_method_raises_an_error`:

```
Boot Error

Something went wrong while loading /Users/tom/Programming/lotus/my_app/config.ru
NoMethodError: undefined method `a_method' for #<Web::Views::ApplicationLayout:0x007fc47d8c7ac8>

/Users/tom/.rvm/gems/ruby-2.1.2/gems/lotus-view-0.3.0/lib/lotus/view/rendering/layout_scope.rb:160:in `rescue in method_missing'
/Users/tom/.rvm/gems/ruby-2.1.2/gems/lotus-view-0.3.0/lib/lotus/view/rendering/layout_scope.rb:157:in `method_missing'
/Users/tom/Programming/lotus/my_app/apps/web/templates/application.html.erb:8:in `block in singleton class'
/Users/tom/Programming/lotus/my_app/apps/web/templates/application.html.erb:-6:in `instance_eval'
/Users/tom/Programming/lotus/my_app/apps/web/templates/application.html.erb:-6:in `singleton class'
/Users/tom/Programming/lotus/my_app/apps/web/templates/application.html.erb:-8:in `__tilt_70240933846960'
/Users/tom/.rvm/gems/ruby-2.1.2/gems/tilt-2.0.1/lib/tilt/template.rb:155:in `call'
/Users/tom/.rvm/gems/ruby-2.1.2/gems/tilt-2.0.1/lib/tilt/template.rb:155:in `evaluate'
/Users/tom/.rvm/gems/ruby-2.1.2/gems/tilt-2.0.1/lib/tilt/template.rb:96:in `render'
/Users/tom/.rvm/gems/ruby-2.1.2/gems/lotus-view-0.3.0/lib/lotus/view/template.rb:39:in `render'
/Users/tom/.rvm/gems/ruby-2.1.2/gems/lotus-view-0.3.0/lib/lotus/layout.rb:128:in `render'
/Users/tom/.rvm/gems/ruby-2.1.2/gems/lotus-view-0.3.0/lib/lotus/view/rendering.rb:104:in `render'
/Users/tom/.rvm/gems/ruby-2.1.2/gems/lotus-view-0.3.0/lib/lotus/view/rendering.rb:236:in `render'
/Users/tom/.rvm/gems/ruby-2.1.2/gems/lotusrb-0.2.1/lib/lotus/rendering_policy.rb:44:in `_render_action'
/Users/tom/.rvm/gems/ruby-2.1.2/gems/lotusrb-0.2.1/lib/lotus/rendering_policy.rb:37:in `_render'
/Users/tom/.rvm/gems/ruby-2.1.2/gems/lotusrb-0.2.1/lib/lotus/rendering_policy.rb:28:in `render'
/Users/tom/.rvm/gems/ruby-2.1.2/gems/lotusrb-0.2.1/lib/lotus/application.rb:214:in `call'
/Users/tom/.rvm/gems/ruby-2.1.2/gems/http_router-0.11.1/lib/http_router.rb:193:in `process_destination_path'
(eval):15:in `call'
/Users/tom/.rvm/gems/ruby-2.1.2/gems/http_router-0.11.1/lib/http_router.rb:288:in `raw_call'
/Users/tom/.rvm/gems/ruby-2.1.2/gems/lotus-router-0.2.1/lib/lotus/routing/http_router.rb:112:in `raw_call'
/Users/tom/.rvm/gems/ruby-2.1.2/gems/http_router-0.11.1/lib/http_router.rb:142:in `call'
/Users/tom/.rvm/gems/ruby-2.1.2/gems/lotus-router-0.2.1/lib/lotus/router.rb:879:in `call'
/Users/tom/.rvm/gems/ruby-2.1.2/gems/lotusrb-0.2.1/lib/lotus/container.rb:20:in `call'
/Users/tom/.rvm/gems/ruby-2.1.2/gems/shotgun-0.9/lib/shotgun/loader.rb:86:in `proceed_as_child'
/Users/tom/.rvm/gems/ruby-2.1.2/gems/shotgun-0.9/lib/shotgun/loader.rb:31:in `call!'
/Users/tom/.rvm/gems/ruby-2.1.2/gems/shotgun-0.9/lib/shotgun/loader.rb:18:in `call'
/Users/tom/.rvm/gems/ruby-2.1.2/gems/rack-1.6.0/lib/rack/lint.rb:49:in `_call'
/Users/tom/.rvm/gems/ruby-2.1.2/gems/rack-1.6.0/lib/rack/lint.rb:37:in `call'
/Users/tom/.rvm/gems/ruby-2.1.2/gems/rack-1.6.0/lib/rack/showexceptions.rb:24:in `call'
/Users/tom/.rvm/gems/ruby-2.1.2/gems/rack-1.6.0/lib/rack/commonlogger.rb:33:in `call'
/Users/tom/.rvm/gems/ruby-2.1.2/gems/rack-1.6.0/lib/rack/content_length.rb:15:in `call'
/Users/tom/.rvm/gems/ruby-2.1.2/gems/rack-1.6.0/lib/rack/handler/webrick.rb:89:in `service'
/Users/tom/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/webrick/httpserver.rb:138:in `service'
/Users/tom/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/webrick/httpserver.rb:94:in `run'
/Users/tom/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/webrick/server.rb:295:in `block in start_thread'
```

**Comment by jodosha** [comment]
@tomkadwill Thank you, this should be more clear. You're right. Do you have time to fix this, or do you want me to do it?

**Comment by tomkadwill** [comment]
@jodosha yes I have time. I haven't looked into it yet but I'll take a stab at solving

**Comment by jodosha** [comment]
@tomkadwill Thank you! :+1:

**Comment by tomkadwill** [comment]
@jodosha I can't reproduce this anymore! Were you able to reproduce it? Were there any fixes add recently that may have fixed this?

**Comment by joneslee85** [comment]
@tomkadwill what version did you try? You could use Github to compare changes between the now version and the version you tried

**Comment by tomkadwill** [comment]
@joneslee85 I was using the following versions:
```
  * lotus-controller (0.3.2)
  * lotus-model (0.2.4)
  * lotus-router (0.2.1)
  * lotus-utils (0.3.4)
  * lotus-validations (0.2.4)
  * lotus-view (0.3.0)
  * lotusrb (0.2.1)
```

I can't reproduce the bug now :confused:. I think it may have been some weirdness trying to include 'lotus/helpers'. If nobody else has experienced this problem then I'll close.

**Comment by joneslee85** [comment]
I have not yet experienced this before so I close the ticket now. Please feel free to re-open if needed.

**Comment by tomkadwill** [comment]
I've now managed to re-produce this issue. Can we re-open? :)

I've created an [example application](https://github.com/tomkadwill/lotus_bug_example) which simulates the problem. essentially, I have a method `universe_age` which contains a method `blah_path`. `blah_path` has not been defined and so the application should, ideally, raise a NoMethodError error for `blah_path`. However, the application raises NoMethodError for `universe_age` instead.

One way to fix the problem is to swap the order of `@scope.__send__` and `@layout.__send__` in the `method_missing` of lotus-view. However, this is probably not the ideal because the application will produce a less informative error in the case the `universe_age` is actually undefined.

Does anyone have any suggestions?

I made a PR with the possible fix. As I said, I don't think this is the ideal fix: https://github.com/lotus/view/pull/63

cc/ @jodosha @joneslee85
